service/custom_logs_post_test.go

```diff
@@ -78,3 +83,17 @@ func Test_CustomLogsPostHandler(t *testing.T) {
 		})
 	}
 }
+
+func Test_CustomLogsPostHandlerInvalidContext(t *testing.T) {
+	handler := service.CustomLogsPostHandler
+
+	r, err := http.NewRequest("POST", "/logs", bytes.NewBuffer([]byte(`{"log_level":"warn","message":"test","data":{"step_id":"test-step-id","tag":"test-tag"}}`)))
+	require.NoError(t, err)
+
+	rr := httptest.NewRecorder()
+	internalServerError := handler(rr, r)
+	expectedInternalErr := "DogStatsD not found in Context"
```

Comment:
NOTE: not the `segmentClient` is missing

Reply:
this is an internal log for DD, coming from GetClientFromContext